File: 2015/day2/day2.js
const fs = require("fs");
const input = fs.readFileSync("input").toString('utf-8').split('\n')
//console.log(input)

//let input = ['4x23x21',  '22x29x19', '11x4x11',  '8x10x5']
//let input = ['2x3x4']

let sum = 0
let inc = 1
for (let line of input) {
  let [l, w, h] = line.split('x').map(Number)

  //Part 1:
  //console.log(l, w, h, Math.min(l*w,w*h,l*h))
  //sum += 2*(l*w + w*h + h*l) + Math.min(l*w,w*h,l*h)

  //Part2:
  sum += 2*Math.min(l+w, l+h, w+h) + l*w*h
}

console.log(`Total surface area required: ${sum}`)

File: 2015/day2/day2.py
with open("input", "r+") as file:
  size = 0
  for line in file:
    l, w, h = map(int, line.strip().split('x'))
    s1 = l*w
    s2 = w*h
    s3 = l*h
    size += 2*s1 + 2*s2 + 2*s3 + min(s1,s2,s3)

print("Part 1:", size)